hat/app/org/hatdex/hat/api/controllers/ContractData.scala

```diff
@@ -0,0 +1,637 @@
+/*
```

Comment:
I don't understand. All of the Contract Data support was moved from RichData to this file.

The net is the addition of Contract Update.  Maybe refactoring more is a suitable new task for refactoring.

Reply:
Sorry i didnt realize that it was copy and pasted code
In that case it could be considered out of scope but i would really like to see it split up if possible and since you are already refactoring it. consider this comment a suggestion/non blocker and i will continue to review. Im assuming the contents of this file are out of scope as well considering its copy and pasted ? In the future to help with this maybe setting scope (i.e what is new code) in the description may help

Reply:
For sure, that's my mistake.

The code in ContractData.scala is pulled out of RichData.scala.  
I then refactored it to extract a uniform pattern for handling contract protected endpoints.

Please see the block of code from 224-288.  These are the entry points for the three Contract protected endpoints.

Reply:
PR context added.

Reply:
Thanks!

Reply:
For future what ive had success with is submitting one PR for moving the code and then another with the refactors, that way it is easy to view the logical changes